pinot-broker/src/main/java/org/apache/pinot/broker/queryquota/HelixExternalViewBasedQueryQuotaManager.java

```diff
@@ -46,21 +55,31 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static org.apache.pinot.spi.utils.CommonConstants.Helix.DATABASE_QUERY_RATE_LIMIT;
+
 
 /**
  * This class is to support the qps quota feature.
- * It depends on the broker source change to update the dynamic rate limit,
- *  which means it only gets updated when a new table added or a broker restarted.
+ * It allows performing qps quota check at table level and database level
+ * For table level check it depends on the broker source change to update the dynamic rate limit,
+ *  which means it gets updated when a new table added or a broker restarted.
+ * For database level check it depends on the broker as well as cluster config change to update the dynamic rate limit,
+ *  which means it gets updated when the default query quota at cluster config is updated or a broker restarted.
```

Comment:
IIUC, the db quota is also updated when a new table is added? as the updating method is called inside 
```
@Transition(from = "OFFLINE", to = "ONLINE")
    public void onBecomeOnlineFromOffline(Message message, NotificationContext context) {
      String tableNameWithType = message.getPartitionName();
```

Reply:
Its is only created and not updated during table creation. The actual creation will only occur during the first table being created for a database (as database creation is implicit and derived from table prefixes).
Even for the above trigger the broker resource change is the source of truth and not the table config.
But I get your point here, will add appropriate doc to mention the creation flow as well